File: src/main/java/org/pixeltime/enchantmentsenhance/events/enchantments/Purge.java
package org.pixeltime.enchantmentsenhance.events.enchantments;

import com.sk89q.worldguard.bukkit.WGBukkit;
import com.sk89q.worldguard.protection.flags.DefaultFlag;
import com.sk89q.worldguard.protection.flags.StateFlag;
import org.bukkit.ChatColor;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.EntityDamageByEntityEvent;
import org.bukkit.potion.PotionEffect;
import org.pixeltime.enchantmentsenhance.manager.SettingsManager;

import java.util.Iterator;

public class Purge implements Listener {
    @EventHandler(priority = EventPriority.MONITOR)
    public void onDamage(final EntityDamageByEntityEvent entityDamageByEntityEvent) {
        final String translateAlternateColorCodes = ChatColor.translateAlternateColorCodes('&', SettingsManager.lang.getString("enchantments." + "purge"));
        if (entityDamageByEntityEvent.getDamager() instanceof Player && entityDamageByEntityEvent.getEntity() instanceof Player) {
            try {
                final Player player = (Player) entityDamageByEntityEvent.getDamager();
                final Player player2 = (Player) entityDamageByEntityEvent.getEntity();
                final int n = (int) (Math.random() * 100.0);
                if (entityDamageByEntityEvent.isCancelled()) {
                    return;
                }
                if (SettingsManager.enchant.getBoolean("allow-worldguard") && WGBukkit.getRegionManager(player2.getWorld()).getApplicableRegions(player2.getLocation()).queryState(null, new StateFlag[]{DefaultFlag.PVP}) == StateFlag.State.DENY) {
                    return;
                }
                if (player.getItemInHand().getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " I") && n < SettingsManager.enchant.getInt("purge.level_I.chance")) {
                    player2.getWorld().strikeLightningEffect(player2.getLocation());
                    final Iterator iterator = player2.getActivePotionEffects().iterator();
                    while (iterator.hasNext()) {
                        player2.removePotionEffect(((PotionEffect) iterator.next()).getType());
                        player2.damage(2.0);
                    }
                }
                if (player.getItemInHand().getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " II") && n < SettingsManager.enchant.getInt("purge.level_II.chance")) {
                    player2.getWorld().strikeLightningEffect(player2.getLocation());
                    final Iterator iterator2 = player2.getActivePotionEffects().iterator();
                    while (iterator2.hasNext()) {
                        player2.removePotionEffect(((PotionEffect) iterator2.next()).getType());
                        player2.damage(2.0);
                    }
                }
                if (player.getItemInHand().getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " III") && n < SettingsManager.enchant.getInt("purge.level_III.chance")) {
                    player2.getWorld().strikeLightningEffect(player2.getLocation());
                    final Iterator iterator3 = player2.getActivePotionEffects().iterator();
                    while (iterator3.hasNext()) {
                        player2.removePotionEffect(((PotionEffect) iterator3.next()).getType());
                        player2.damage(3.0);
                    }
                }
                if (player.getItemInHand().getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " IV") && n < SettingsManager.enchant.getInt("purge.level_IV.chance")) {
                    player2.getWorld().strikeLightningEffect(player2.getLocation());
                    final Iterator iterator4 = player2.getActivePotionEffects().iterator();
                    while (iterator4.hasNext()) {
                        player2.removePotionEffect(((PotionEffect) iterator4.next()).getType());
                        player2.damage(4.0);
                    }
                }
            } catch (Exception ex) {
            }
        }
    }
}

File: src/main/kotlin/org/pixeltime/enchantmentsenhance/event/enchantment/Purge.kt
/*
 *     Copyright (C) 2017-Present HealPotion
 *
 *     This program is free software: you can redistribute it and/or modify
 *     it under the terms of the GNU General Public License as published by
 *     the Free Software Foundation, either version 3 of the License, or
 *     (at your option) any later version.
 *
 *     This program is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *     GNU General Public License for more details.
 *
 *     You should have received a copy of the GNU General Public License
 *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
 *
 */

package org.pixeltime.enchantmentsenhance.event.enchantment

import com.sk89q.worldguard.bukkit.WGBukkit
import com.sk89q.worldguard.protection.flags.DefaultFlag
import com.sk89q.worldguard.protection.flags.StateFlag
import org.bukkit.ChatColor
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.Listener
import org.bukkit.event.entity.EntityDamageByEntityEvent
import org.bukkit.potion.PotionEffect
import org.pixeltime.enchantmentsenhance.manager.IM
import org.pixeltime.enchantmentsenhance.manager.KM
import org.pixeltime.enchantmentsenhance.manager.SettingsManager

class Purge : Listener {
    @EventHandler(priority = EventPriority.MONITOR)
    fun onDamage(entityDamageByEntityEvent: EntityDamageByEntityEvent) {
        val translateAlternateColorCodes = ChatColor.translateAlternateColorCodes('&', SettingsManager.lang.getString("enchantments." + "purge"))
        if (entityDamageByEntityEvent.damager is Player && entityDamageByEntityEvent.entity is Player) {

            val player = entityDamageByEntityEvent.damager as Player
            val player2 = entityDamageByEntityEvent.entity as Player
            if (entityDamageByEntityEvent.isCancelled) {
                return
            }
            if (SettingsManager.enchant.getBoolean("allow-worldguard") && WGBukkit.getRegionManager(player2.world).getApplicableRegions(player2.location).queryState(null, DefaultFlag.PVP) == StateFlag.State.DENY) {
                return
            }
            try {
                val armorContents = player.inventory.armorContents + IM.getAccessorySlots(player)
                for (itemStack in armorContents) {
                    if (itemStack.hasItemMeta() && itemStack.itemMeta.hasLore()) {
                        val level = KM.getLevel(translateAlternateColorCodes, itemStack.itemMeta.lore)
                        if (level > 0 && (Math.random() * 100.0).toInt() < SettingsManager.enchant.getInt("purge.$level.chance")) {
                            player2.world.strikeLightningEffect(player2.location)
                            val iterator = player2.activePotionEffects.iterator()
                            while (iterator.hasNext()) {
                                player2.removePotionEffect((iterator.next() as PotionEffect).type)
                                player2.damage(2.0)
                            }
                        }
                    }
                }
            } catch (ex: Exception) {
            }

        }
    }
}
